### PR TITLE
Added In-place heap sort and Max Priority Queue.

### PR DESCRIPTION
**Pull request by Tanmay2510**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Comment by Tanmay2510** [comment]
Can you review the new pull request @HarshCasper @atarax665 ?

File: Java/README.md [hunk 1]
```diff
@@ -107,7 +108,10 @@ _add list here_
 - [Tree Traversal(preorder traversal)](ds/preOrderTraversal.java)
 - [Tree Traversal(postorder traversal)](ds/postorder_Traversal.java)
 - [Priority Queue](ds/PriorityQueueEg.java)
+
+=======
```
Comment (atarax665) [inline Java/README.md]: Remove all these irrelevant lines that have been pushed due to improper use of git commands.

File: Java/ds/maxPriorityQueue.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
+//Implementing the class for Max Priority Queue which includes the required functions......
```
Comment (atarax665) [inline Java/ds/maxPriorityQueue.java]: Enclose the description in multi-line comment.

**Review by atarax665 — CHANGES_REQUESTED** [review]
- Add driver code in both the programs.
- Update README for Max Priority Queue too.

File: Java/ds/maxPriorityQueue.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+          System.out.println();
+         }
+	
+     public static void main(String args[]) {
+
+       ArrayList<Integer> array = new ArrayList<Integer>();
+       int size = array.size();
+ 
+       Heap h = new Heap();
+       h.insert(array, 6);
```
Comment (atarax665) [inline Java/ds/maxPriorityQueue.java]: Take dynamic inputs.

File: Java/sort/InPlaceHeapsort.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      {
+           int n = A.length;
+           for (int i=0; i<n; ++i)
+            System.out.print(A[i]+" ");
+         System.out.println();
+      }
+
+   public static void main(String args[])
+     {
+          int A[] = {2, 6, 8, 5, 4, 3};
```
Comment (atarax665) [inline Java/sort/InPlaceHeapsort.java]: Take dynamic inputs.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by Tanmay2510** [comment]
Please review it @atarax665 @HarshCasper .

File: Java/ds/maxPriorityQueue.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+          System.out.println();
+         }
+	
+     public static void main(String args[]) {
+                st = new StringTokenizer(br.readLine());
+		PQ pq = new PQ();
+		int choice = Integer.parseInt(st.nextToken());
+		while(choice != -1) {
+			switch(choice) {
+				case 1 : // insert
```
Comment (atarax665) [inline Java/ds/maxPriorityQueue.java]: If you are making a menu driven program then make sure you prompt user all the available choices as a program like this should be much more interactive.

File: Java/ds/maxPriorityQueue.java [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+					System.out.println(pq.isEmpty());
+				default :
+						return;
+			}
+			choice = Integer.parseInt(st.nextToken());
+      
+         }
+}
+/*
+ Sample Input
```
Comment (atarax665) [inline Java/ds/maxPriorityQueue.java:133]: Sample I/O should be exactly what once you see on the terminal when they run this program. Please demonstrate all the functionalities of the program.

File: Java/ds/maxPriorityQueue.java [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+	}
+     void printArray(ArrayList<Integer> array, int size) {
+        for (Integer i : array) {
+           System.out.print(i + " ");
+           }
+          System.out.println();
+         }
+	
+     public static void main(String args[]) {
+                st = new StringTokenizer(br.readLine());
```
Comment (atarax665) [inline Java/ds/maxPriorityQueue.java]: There is inconsistent indentation throughout the code, please use any java formatter to format your code.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by atarax665** [comment]
You have not resolved all the conversations. Request review only when you are done with the requested changes.

**Pull request closed without merging**